src/main.js

```diff
@@ -117,6 +137,19 @@ class PlatformIOVSCodeExtension {
     ));
   }
 
+  terminateUploadTask() {
+    if(this.inUploadTask) {
+      this.inUploadTask = false;
+      return vscode.commands.executeCommand("workbench.action.tasks.terminate").then(() => {
```

Comment:
It terminates currently running foreground task, of which there can only be one I think. The "Upload and Monitor" task keeps running in the foreground as long as the serial port is connected.